File: ed2/gara4/programma/att/programma.c
#include <stdio.h>
#include <assert.h>

#define MAXN 1000000
#define MAXM 100000

typedef struct {
    int x, y;
} point;

point passeggia(int N, int M, char P[], int S[], int E[]) {
    // Mettete qui il codice della soluzione
    point p;
    p.x = 42;
    p.y = 42;
    return p;
}


char P[MAXN+2];
int S[MAXM], E[MAXM];

int main() {
    FILE *fr, *fw;
    int N, M, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d %d\n", &N, &M));
    assert(1 == fscanf(fr, "%s", P));
    for(i=0; i<M; i++)
        assert(2 == fscanf(fr, "%d %d", &S[i], &E[i]));

    point p = passeggia(N, M, P, S, E);
    fprintf(fw, "%d %d\n", p.x, p.y);
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/gara4/programma/att/programma.pas
type
	point = record
		x, y : longint;
	end;

function passeggia(N, M: longint; var P: array of char; var S, E: array of longint): point;
begin
    (* Mettete qui il codice della soluzione *)
    passeggia.x := 42;
    passeggia.y := 42;
end;

const
    MAXN = 1000000;
    MAXM = 100000;
var
    N, M, i : longint;
    P       : array[0..MAXN+1] of char;
    S, E    : array[0..MAXM-1] of longint;
    res     : point;
    fr, fw  : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, M);
    readln(fr, P);
    for i:=0 to M-1 do
        readln(fr, S[i], E[i]);

    res := passeggia(N, M, P, S, E);
    writeln(fw, res.x, ' ', res.y);
    close(fr);
    close(fw);
end.
